### PR TITLE
Modify assemblies to be variants instead of product with few minor fixes

### PR DESCRIPTION
**Pull request by nishant-cyro**
This PR allows to add parts on variant level instead of just product level.

Please refer - https://github.com/spree-contrib/spree-product-assembly/issues/132

### CODE REVIEW

File: spec/models/spree/order_inventory_spec.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ module Spree
 
       let(:bundle) { create(:product) }
 
-      before { bundle.parts.push [guitar, bass] }
+      before { bundle.master.parts.push([guitar, bass]); bundle.reload }
```
Comment (vinniefranco) [inline spec/models/spree/order_inventory_spec.rb]: Thoughts on breaking this from `{` block syntax to `do` `end` syntax? Not a big fan of the semicolon approach

Comment (Mafi88) [inline spec/models/spree/order_inventory_spec.rb]: I agree :)

**Comment by Mafi88** [comment]
@nishant-cyro Does not work with Spree 3.1 (not compatible with sprockets-rails 3). Please add assets to precompilation.

**Comment by nishant-cyro** [comment]
@Mafi88 -  I have added the spinner.gif image in asset precompile in engine.rb.

```
initializer 'spree.assets.precompile', group: :all do |app|
  app.config.assets.precompile += %w( spinner.gif )
end
```

Is there anything else I missed in your comment? Please let me know.

**Comment by Mafi88** [comment]
@nishant-cyro Right, my bad. Works properly. Would be great if select for variant could be more spree-like.
And could you please address @vinniefranco comment?

**Comment by nishant-cyro** [comment]
@Mafi88 - Where would you like to have the changes for variant selection in admin end or user end.

Also, it will be good if you specify it more. Thanks in Advance :)

@vinniefranco  - Done with the change you specified.

**Comment by Mafi88** [comment]
@nishant-cyro In admin panel. When you choose variant for part, there is a standard non-spree selectbox. Maybe something like this would be better? https://github.com/spree/spree/blob/master/backend/app/views/spree/admin/products/_form.html.erb#L144

**Comment by nishant-cyro** [comment]
@Mafi88 - Done with the changes. Can you rerun the build again?

**Comment by Mafi88** [comment]
@nishant-cyro Actually, I can't ;/

**Pull request closed without merging**